### PR TITLE
Minimum commission for validators

### PR DESCRIPTION
**Pull request by kianenigma**
This PR will add a new configurable storage item to staking that allows setting the minimum amount of commission for validators.

As of now, this is not enforced in already existing validators at all. Only those that wish to become one, or update their prefs, will need to comply with this limit, if set. The default is `0%`, which means all values comply with it.  

Renames the previous `set_staking_limits` to `set_staking_configs` and reuses that as the governance entry point.

---

A technical dilemma is that `set_staking_configs` is becoming increasingly bloated. Ideally, I can see an instantiable `pallet-configurations` that can contain these configurations, and provide easy transactions to set them. As of now, if you want to tweak just one of these configs, you have to set all of them. Any mistake in the process can be fatal. I think it is fair to say whoever is calling that function must be super careful. But, long term we can improve this. 

Polkadot companion: https://github.com/paritytech/polkadot/pull/4413
skip check-dependent-cumulus

### CODE REVIEW

**Comment by burdges** [comment]
Should we grandfather in existing validators?  Or just compute everyone's commission as `max(their_commission,minimum_commission)`?

**Comment by kianenigma** [comment]
> Should we grandfather in existing validators?  Or just compute everyone's commission as `max(their_commission,minimum_commission)`?

I don't think it is a good idea to change the preferences on the fly. Enforcing them upon "validate" should be enough to ensure all validators eventually comply. 

That being said, we can wrote a migration for that as well. But it will need to be a separate PR.

File: frame/staking/src/pallet/mod.rs [hunk 1]
```diff
@@ -210,6 +210,12 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type MinValidatorBond<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
 
+	/// The minimum amount of commission that validators can set.
+	///
+	/// If non-existent, no limit exists.
```
Comment (kianenigma) [inline frame/staking/src/pallet/mod.rs:215]:
```suggestion
	/// If set to `0`, no limit exists.
```

File: frame/staking/src/weights.rs [hunk 2]
```diff
@@ -178,12 +173,9 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
...
 	// Storage: BagsList CounterForListNodes (r:1 w:1)
-	fn nominate(n: u32, ) -> Weight {
-		(82_389_000 as Weight)
-			// Standard Error: 14_000
-			.saturating_add((5_597_000 as Weight).saturating_mul(n as Weight))
-			.saturating_add(T::DbWeight::get().reads(12 as Weight))
-			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(n as Weight)))
+	fn nominate(_n: u32, ) -> Weight {
+		(119_946_000 as Weight)
+			.saturating_add(T::DbWeight::get().reads(21 as Weight))
```
Comment (shawntabrizi) [inline frame/staking/src/weights.rs:178]: hopefully these changes are fixed after the merge master

**Review by shawntabrizi — APPROVED** [review]

**Review by emostov — APPROVED** [review]

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**